webapp/src/js/enketo/widgets/phone-widget.js

```diff
@@ -5,38 +5,49 @@ const $ = require( 'jquery' );
 const phoneNumber = require('@medic/phone-number');
 require( 'enketo-core/src/js/plugins' );
 
-// Set up enketo validation for `phone` input type
+const isContactPhoneValid = (settings, fieldValue) => {
+  if (phoneNumber.validate(settings, fieldValue)) {
+    return true;
+  }
+
+  console.error( 'invalid phone number: "' + fieldValue + '"' );
+  return false;
+};
+
+const getContactIdsForPhone = (phoneNumber) => window.CHTCore.DB
+  .get()
+  .query('medic-client/contacts_by_phone', { key: phoneNumber })
+  .then(results => results.rows.map(row => row.id));
+
+const isContactPhoneUnique = async (settings, fieldValue) => {
+  const normalizedNumber = phoneNumber.normalize(settings, fieldValue);
+  const contactIds = await getContactIdsForPhone(normalizedNumber);
+  const contactBeingEdited = $('#contact-form').attr('data-editing');
+  if (!contactIds.length || contactBeingEdited && contactIds.includes(contactBeingEdited)) {
+    return true;
+  }
+
+  console.error('phone number not unique: "' + fieldValue + '"');
+  return false;
+};
+
+
+// Set up enketo validation for data types
 FormModel.prototype.types.tel = {
-  validate: function( fieldValue ) {
-    return window.CHTCore.Settings
-      .get()
-      .then( function( settings ) {
-        if ( !phoneNumber.validate( settings, fieldValue ) ) {
-          throw new Error( 'invalid phone number: "' + fieldValue + '"' );
```

Comment:
Looking at the implementation of the other [Enketo Type validations](https://github.com/enketo/enketo/blob/351944c491cd59c6a0574d40393a4bb9fca7f81e/packages/enketo-core/src/js/types.js), it seems that none of them are deliberately throwing an error.  Instead they just return `false` when the validation fails.  The main behavior difference that I could find between returning `false` and just throwing an error is that an `invalidated` event is not properly emitted for the input when an error is thrown.  

To remain more consistent with existing Enketo behavior, I have updated the validation logic to just return `false` instead of throwing errors.